TABTestKit/Classes/Contexts/NavigationContext.swift

```diff
@@ -53,11 +53,44 @@ public extension NavigationContext {
 	///
 	/// - Parameter dismissableThings: One or more Dismissable things. Typically, this would be a Screen that conforms to Dismissable.
 	func dismiss(_ dismissableThings: Dismissable...) {
-		guard !dismissableThings.isEmpty else { XCTFatalFail("You must provide at least one Dismissable thing to dismizs!") }
+		guard !dismissableThings.isEmpty else { XCTFatalFail("You must provide at least one Dismissable thing to dismiss!") }
 		dismissableThings.forEach {
 			$0.await()
 			$0.dismiss()
 		}
 	}
+
+  /// Opens the specified URL from the test runner app.
+  /// This means you can use this function for testing deep linking, as well as opening web urls in Safari.
+  /// This function will first show the Home Screen if necessary, launch the test runner (after finding the correct
+  /// page with the runner) and then open the URL.
+  ///
+  /// It is your responsibility/choice if you want to leave your app in the current state before the test runner
+  /// opens the URL, or whether you want to completely terminate your app before hand.
+  ///
+  /// This leaves you free to test deep linking behaviour from the current state of the app.
+  /// or a fresh launch of the app.
+  ///
+  /// Since you can pass any URL to this function, you could pass an app-specific url like `myapp://deep-link`, or
+  /// a regular HTTP link like `https://apple.com`.
+  ///
+  /// HTTP links are opened in Safari by default, unless you have an app installed that can intercept them using
+  /// Universal Links.
+  ///
+  /// - Parameter url: The URL to open.
+  func open(_ url: URL) {
+   springboard.activate()
+    if !Icon.testRunner.underlyingXCUIElement.isHittable {
+      XCUIDevice.shared.press(.home) // Ensure we're on the first page of the home screen
+      sleep(1)
+      while !Icon.testRunner.underlyingXCUIElement.isHittable {
+        springboard.swipeLeft()
+        sleep(1)
+      }
+    }
+    Icon.testRunner.tap()
+    sleep(1)
+    UIApplication.shared.open(url)
+  }
```

Comment:
So basically what's happening here is:
- Activate Springboard (which shows the Home Screen if it's not already showing)
- If the testRunner icon isn't already showing, press home so that we go to the first page of icons
- Swipe left until the testRunner icon is hittable and tap it
- Open the URL from the test runner's shared UIApplication instance (this **isn't** the main app's UIApplication instance)

I tried creating an XCUIApplication instance for the test runner specifically using the bundle ID but doesn't look like XCUI is allowed to interact with it that way

Reply:
Nice little workaround. The ability to control the simulator outside of the app under test is still so immature but at least it's fairly limited in size so you have fairly robust ways to navigate it.

Reply:
Just looking at adding an example test for this